xformers/components/attention/csrc/cuda/mem_eff_attention/attention_forward_generic.cu

```diff
@@ -62,19 +62,49 @@
   }
 
 namespace {
+/*
+  There are 2 modes for using this function.
+  (Mode BMK) `num_heads=1` with the heads coalesced into the batch dimension
```

Comment:
this is not actually a requirement, right? your kernel can take a `[B, L, num_heads, head_dim]` dense tensor and work right?